apollo-router/src/query_planner/dual_query_planner.rs

```diff
@@ -143,12 +143,10 @@ impl BothModeComparisonJob {
                     tracing::debug!("JS and Rust query plans match{operation_desc}! 🎉");
                 } else {
                     tracing::debug!("JS v.s. Rust query plan mismatch{operation_desc}");
-                    if let Some(formatted) = &js_plan.formatted_query_plan {
-                        tracing::debug!(
-                            "Diff of formatted plans:\n{}",
-                            render_diff(&diff::lines(formatted, &rust_plan.to_string()))
-                        );
-                    }
+                    tracing::debug!(
+                        "Diff of formatted plans:\n{}",
+                        diff_plan(&js_plan, &rust_plan)
+                    );
```

Comment:
The rendered diff wouldn't be as accurate when comparing formatted plans, since both JS and Rust don't show all the information that could have diverged.

Reply:
It's good enough for now. We can improve the diff output as we go.

Clippy suggestion:
```suggestion
                    tracing::debug!(
                        "Diff of formatted plans:\n{}",
                        diff_plan(js_plan, rust_plan)
                    );
```